.github/copilot-instructions.md

```
@@ -1,187 +1,78 @@
+| `pio run -e <env>` | Build firmware for a hardware target | 15–20 min |
+
+**Always run `npm run build` before `pio run`.** The web UI build generates `wled00/html_*.h` header files required by firmware compilation.
+
+Common firmware environments: `esp32_4MB_V4_M`, `esp32_16MB_V4_S_HUB75`, `esp32S3_8MB_PSRAM_M_qspi`, `esp32_16MB_V4_M_eth`,`esp32dev_compat`, `esp8266_4MB_S` (deprecated)
+
+For detailed build timeouts, development workflows, troubleshooting, and validation steps, see [agent-build-instructions.md](agent-build-instructions.md).
+
+## Repository Structure
 
-### Repository Structure
 ```
```

markdownlint-cli2 (0.22.0)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)